include/nbl/ext/Scan/Scan.h

```diff
@@ -0,0 +1,190 @@
+#ifndef _NBL_EXT_SCAN_INCLUDED_
+
+#include "nabla.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace Scan
+{
+
+typedef uint32_t uint;
+#include "nbl/builtin/glsl/ext/Scan/parameters_struct.glsl"
+
+template <typename T>
+class Scan final : public core::IReferenceCounted
+{
+public:
+	_NBL_STATIC_INLINE_CONSTEXPR uint32_t DEFAULT_WORKGROUP_SIZE = 256u;
+
+	enum Operator : uint32_t
+	{
+		AND = 1 << 0,
+		XOR = 1 << 1,
+		OR  = 1 << 2,
+		ADD = 1 << 3,
+		MUL = 1 << 4,
+		MIN = 1 << 5,
+		MAX = 1 << 6,
```

Comment:
not sure why then need to be separate bits, its not like you can do more than 1 at once

Reply:
you can't, its arbitrary

Reply:
not a huge issue, it just seemed weird to me... that's all